packages/core/src/codewhisperer/service/securityScanHandler.ts

```diff
@@ -109,19 +111,34 @@ function mapRawToCodeScanIssue(issue: RawCodeScanIssue): CodeScanIssue {
     }
 }
 
-function mapToAggregatedList(codeScanIssueMap: Map<string, RawCodeScanIssue[]>, json: string) {
+function mapToAggregatedList(
+    codeScanIssueMap: Map<string, RawCodeScanIssue[]>,
+    json: string,
+    editor: vscode.TextEditor | undefined,
+    scope: CodeWhispererConstants.CodeAnalysisScope
+) {
     const codeScanIssues: RawCodeScanIssue[] = JSON.parse(json)
     codeScanIssues.forEach(issue => {
-        if (codeScanIssueMap.has(issue.filePath)) {
-            const list = codeScanIssueMap.get(issue.filePath)
-            if (list === undefined) {
-                codeScanIssueMap.set(issue.filePath, [issue])
+        let shouldShow = true
+
+        if (scope === CodeWhispererConstants.CodeAnalysisScope.FILE && editor) {
+            shouldShow = issue.codeSnippet.every(code => {
```

Comment:
IIRC doesn't the code snippet include a few lines before and after? If so we need to check only the impacted line numbers

Reply:
Right now,
I am checking all the text provided in the code snippet. Should I restrict this to only the impacted line numbers ?